test/masq.test.js

```diff
@@ -536,6 +435,34 @@ describe('Test login procedure', () => {
 // TODO add tests for connect-disconnect-connect
 
 describe('Test data access and input', () => {
+  test('operations should fail if masq is not connected', async () => {
+    const functions = [ 'watch' ]
+    const promises = [
+      masq.get('key'),
+      masq.put('key', 'value'),
+      masq.del('key'),
+      masq.list('/')
+    ]
+
+    expect.assertions(promises.length + functions.length)
+
+    promises.forEach(async (p) => {
+      try {
+        await p
+      } catch (e) {
+        expect(e.type).toBe(ERRORS.NOT_CONNECTED)
+      }
```

Comment:
In general, when we have expect in catch, instead of expect.assertions with the number of assertions we could use the following :
let err = undefined
try {
  await p
} catch (e) {
  err = e
}
expect(err.type).toBe(ERRORS.NOT_CONNECTED)

What do you think ?

Reply:
Not in this case but in general, it avoids having to specify a constant as the number of expect which can change if we change when we edit the test

Reply:
There is also this syntax: https://jestjs.io/docs/en/tutorial-async.html#rejects, They specified expect.assertions in the example but don't think this is necessary ? What to you think about this

Reply:
I'm unable to make it work. I will use your solution @sujeebant thanks

Reply:
I find It's too verbose to surround with try catch throught, if you find another solution ;)